runner/src/internal.rs

```diff
@@ -97,24 +97,76 @@ pub struct BuildServer {
     pub coverage: bool,
 }
 
+#[derive(Clone)]
+/// A construct to keep track of the status of the execution. It only cares about the top-level
+/// steps.
+pub struct Status {
+    failed: usize,
+    passed: usize,
+    remaining: usize,
```

Comment:
This terminology seems unnecessarily different from the OK / Error terminology that is already used throughout the code and the logs. FWIW I avoided the pass / fail terminology because not all steps are "tests".